node/src/database/db_migrations.rs

```diff
@@ -507,50 +511,54 @@ mod tests {
         let iterator = list_of_updates.iter();
         let iterator_shifted = list_of_updates.iter().skip(1);
         iterator.zip(iterator_shifted).for_each(|(first, second)| {
-            if compare_two_numbers(first.old_version(), second.old_version()).not() {
+            if assert_on_two_numbers_tight_continuity(first.old_version(), second.old_version())
+                .not()
+            {
                 panic!("The list of updates for the database is not ordered properly")
             }
         });
     }
 
-    fn compare_two_numbers(first: usize, second: usize) -> bool {
-        first.le(&second)
+    fn assert_on_two_numbers_tight_continuity(first: usize, second: usize) -> bool {
+        (first + 1) == second
     }
 
     #[test]
-    fn list_of_existing_updates_is_correctly_ordered() {
-        let _ = list_validation_check(DbMigratorReal::list_of_existing_updates());
+    fn list_of_approved_updates_is_correctly_ordered() {
+        let _ = list_validation_check(DbMigratorReal::list_of_approved_updates());
         //success if no panicking
     }
 
     #[test]
-    fn list_of_existing_updates_does_not_end_with_version_higher_than_the_current_version() {
-        let last_entry = DbMigratorReal::list_of_existing_updates()
+    fn list_of_approved_updates_does_not_end_with_version_higher_than_the_current_version() {
+        let last_entry = DbMigratorReal::list_of_approved_updates()
             .into_iter()
             .last();
 
         let result = last_entry.unwrap().old_version();
 
-        assert!(compare_two_numbers(result, CURRENT_SCHEMA_VERSION))
+        assert!(assert_on_two_numbers_tight_continuity(
+            result,
+            CURRENT_SCHEMA_VERSION
+        ))
     }
 
     #[test]
     fn migrate_semi_automated_returns_an_error_from_update_schema_version() {
         let update_schema_version_params_arc = Arc::new(Mutex::new(vec![]));
-        let mut migration_record = DBMigrationRecordMock::default().migrate_result(Ok(()));
+        let mut migration_record = DBMigrationRecordMock::default()
+            .old_version_result(4)
+            .migrate_result(Ok(()));
         let migration_utilities = DBMigrationUtilitiesMock::default()
             .update_schema_version_result(Err(Error::InvalidQuery))
             .update_schema_version_params(&update_schema_version_params_arc);
 
-        let result = DbMigratorReal::migrate_semi_automated(
-            &mut migration_record,
-            5.to_string(),
-            &migration_utilities,
-        );
+        let result =
+            DbMigratorReal::migrate_semi_automated(&mut migration_record, &migration_utilities);
 
         assert_eq!(result, Err(Error::InvalidQuery));
         let update_schema_version_params = update_schema_version_params_arc.lock().unwrap();
-        assert_eq!(*update_schema_version_params, vec!["5".to_string()])
+        assert_eq!(*update_schema_version_params, vec![5])
```

Comment:
If the migration from 4 to 5 failed, should we really be updating the schema version to 5, or should we instead leave it at 4?

Reply:
This looks different than it is. In the bigger context, this function's failure (update_schema_version()) means that either writing the new version hasn't been completed and thus there is not a waiting change to the db, updating the schema version, or , and what is more important, any failure at that point just means that the db won't modify its state, because we haven't committed any changes we won't because the failure doesn't allow to step up to the final step.
I think it's completely okay. Only this test may confuse somebody, but I don't know how to avoid it without writing unnecessary lines.